neal/sampler.py

```diff
@@ -239,21 +265,56 @@ def sample(self, bqm, beta_range=None, num_reads=None, num_sweeps=1000,
         if interrupt_function and not callable(interrupt_function):
             raise TypeError("'interrupt_function' should be a callable")
 
+        if num_sweeps_per_beta < 1:
```

Comment:
Do you also want to prevent non-integer values of `num_sweeps_per_beta` here? The current code will accept something like `sampler.sample_ising(h, J, beta_schedule_type="custom", beta_schedule=[1, 2], num_sweeps_per_beta=2.34)` without an error.

Reply:
Yes, I'll add an Integral check as well.